### ISSUE TITLE
SkeletonTableViewDataSource numberOfRowsInSection never called

### ISSUE THREAD

**Issue by Erumaru** [Juanpe/SkeletonView#269]
⚠️ Please fill out this template when filing an issue.

### What did you do?

Created `UITableView` with `isSkeletonable` set to `true`. All cells and its subview also `isSkeletonable` set to `true`. Implemented SkeletonTableViewDataSource and numberOfRowsInSection in `MyViewController` class. Set `UITableView`'s dataSource to `MyViewController`. Called `showSkeleton()`.

### What did you expect to happen?

I expected `UITableView` to use 
```
func collectionSkeletonView(_ skeletonView: UITableView, numberOfRowsInSection section: Int) -> Int
``` 

### What happened instead?

It always uses
```
func tableView(_ tableView: UITableView, numberOfRowsInSection section: Int) -> Int
``` 

### Steps to reproduce the behavior

Steps mentioned in *what did you do*

### SkeletonView Environment

**SkeletonView version:** 1.8.6
**Xcode version:** 11.3.1
**Swift version:** 5.0

**Comment by paulanatoleclaudot-betclic** [comment]
showSkeleton should be called on your collection/tableView instead of your view :)

**Comment by aipinn** [comment]
Me too.

**Comment by Erumaru** [comment]
@paulanatoleclaudot-betclic thanks for your response, but I call it on collection view.

**Comment by tadeha** [comment]
Me too. It doesn't calculate how many cells need to populate the whole table view.

**Comment by aipinn** [comment]
Boys, I resolved it. I used vc addsubview tableView.

1. First, you must set vc.view.isSkeletonable = true
2. Second, for tableView must use estimatedRowHeight.

```   tableView.estimatedRowHeight = 130
        tableView.rowHeight = UITableView.automaticDimension
```

**Comment by aipinn** [comment]
Boys, I resolved it. I used vc addsubview tableView.

First, you must set vc `self.view.isSkeletonable = true`
Second, for tableView must use estimatedRowHeight.
```
tableView.estimatedRowHeight = 130
tableView.rowHeight = UITableView.automaticDimension
```

**Comment by finetuning-mg** [comment]
Same problem here with 1.8.6 and 1.8.7. Fixed by reverting to version 1.8.5.
I am on Xcode 11.3.1 and Swift 5.1

**Comment by mkval** [comment]
Alright. So if you're subclassing directly from `UITableViewController`, you somehow need to explicitly set this controller as the `dataSource` of the `tableView`. 

Without this, when you set a breakpoint at line 35 of the file `UITableView+CollectionSkeleton`, `guard` always fails because it detects that your `dataSource` is of type `_UIFilteredDataSource` and not `SkeletonTableViewDataSource`.

**Comment by mkval** [comment]
Got the same issue. But instead of composing my own custom TableViewController (`UIViewController` + `UITableView`), I instead opted to subclassing `UITableViewController`. So somehow, you need to explicitly set this controller as the `dataSource` of the `tableView` to make SV work. 

```
override func viewDidLoad() {
  super.viewDidLoad()
  ...
  tableView.dataSource = self
  ...
}
```

Without this, when you set a breakpoint at line 35 of the file `UITableView+CollectionSkeleton`, `guard` always fails because it detects that your `dataSource` is of type `_UIFilteredDataSource` and not `SkeletonTableViewDataSource`.

Here's my spec: _SkeletonView_: 1.8.7 | _Swift_: 5 | _Xcode_: 11.2.1

**Comment by XiaoxiaYao** [comment]
> Boys, I resolved it. I used vc addsubview tableView.
> 
> First, you must set vc `self.view.isSkeletonable = true`
> Second, for tableView must use estimatedRowHeight.
> 
> ```
> tableView.estimatedRowHeight = 130
> tableView.rowHeight = UITableView.automaticDimension
> ```

Yep. When you find all the cells are not displaying with `showSkeleton()`, please set the `estimatedRowHeight`. Otherwise, they won't show it!!! I think it's better to put this into README. @Juanpe Maybe a Q&A part.

**Comment by Juanpe** [comment]
Hi @XiaoxiaYao, you're right. Actually, this info is already in the [README](https://github.com/Juanpe/SkeletonView#-collections) file:

![Screenshot 2021-08-25 at 18 57 24](https://user-images.githubusercontent.com/1409041/130833313-2237e5d0-34c7-4917-a7dd-13ecb1fd12ce.png)

But, yes, I agree with you that a Q&A or FAQ section could be very useful. Feel free to create it :)
